addons/project/models/project_task_recurrence.py

```diff
@@ -81,13 +81,15 @@ def _get_recurrence_delta(self):
 
     def _create_next_occurrence(self, occurrence_from):
         self.ensure_one()
-        if self.repeat_type == 'until' and fields.Date.today() > self.repeat_until:
-            return
         # Prevent double mail_followers creation
         self = self.with_context(mail_create_nosubscribe=True)
-        self.env['project.task'].sudo().create(
-            self._create_next_occurrence_values(occurrence_from)
+        create_values = self._create_next_occurrence_values(occurrence_from)
+        exceeds_limit_date = any(
+            create_values[field] and create_values[field].date() > self.repeat_until
+            for field in self._get_recurring_fields_to_postpone()
         )
+        if not (self.repeat_type == 'until' and exceeds_limit_date):
```

Comment:
```suggestion
        if not (self.repeat_type == 'until' and create_values['date_deadline'] > self.repeat_until):
```
Isn't that enough? :slightly_smiling_face:

Reply:
Done